### PR TITLE
Add metrics for execution time of RunOnce

### PR DESCRIPTION
**Pull request by kgolab**
Covers Recommender's and Updater's RunOnce.

@bskiba, please have a look.

### CODE REVIEW

File: vertical-pod-autoscaler/pkg/recommender/routines/recommender.go [hunk 1]
```diff
@@ -106,25 +106,34 @@ func (r *recommender) updateVPAs() {
...
+		timer.ObserveStep("StoreCheckpoints")
 		if time.Now().Sub(r.lastCheckpointGC) > r.checkpointsGCInterval {
 			r.lastCheckpointGC = now
 			r.clusterStateFeeder.GarbageCollectCheckpoints()
+			timer.ObserveStep("CheckpointsGC")
 		}
 	}
 	if time.Now().Sub(r.lastAggregateContainerStateGC) > AggregateContainerStateGCInterval {
 		r.clusterState.GarbageCollectAggregateCollectionStates(now)
+		timer.ObserveStep("AggregatesGC")
```
Comment (bskiba) [inline vertical-pod-autoscaler/pkg/recommender/routines/recommender.go]: AggregateStatesGC

File: vertical-pod-autoscaler/pkg/utils/metrics/recommender/recommender.go [hunk 2]
```diff
@@ -52,6 +52,9 @@ var (
 			Buckets:   []float64{10.0, 20.0, 30.0, 40.00, 50.0, 60.0, 90.0, 120.0, 150.0, 180.0, 240.0, 300.0, 600.0, 900.0, 1800.0},
 		},
 	)
+
+	functionLatency = metrics.CreateExecutionTimeMetric(metricsNamespace,
+		"Time spent in various part of VPA Recommender main loop.")
```
Comment (bskiba) [inline vertical-pod-autoscaler/pkg/utils/metrics/recommender/recommender.go]: s/part/parts/

File: vertical-pod-autoscaler/pkg/utils/metrics/updater/updater.go [hunk 3]
```diff
@@ -34,11 +34,20 @@ var (
 			Help:      "Number of Pods evicted by Updater to apply a new recommendation.",
 		},
 	)
+
+	functionLatency = metrics.CreateExecutionTimeMetric(metricsNamespace,
+		"Time spent in various part of VPA Updater main loop.")
```
Comment (bskiba) [inline vertical-pod-autoscaler/pkg/utils/metrics/updater/updater.go]: s/part/parts/

**Comment by bskiba** [comment]
Just nits

**Comment by kgolab** [comment]
Removed the nits (including one more found elsewhere) & squashed.

**Pull request merged**